lib/extensions.js

```diff
@@ -40,6 +40,34 @@ module.exports = function(realmConstructor) {
     setConstructorOnPrototype(realmConstructor.Results);
     setConstructorOnPrototype(realmConstructor.Object);
 
+    //Add async open API
+    Object.defineProperties(realmConstructor, getOwnPropertyDescriptors({
+        open(config) {
+            return new Promise((resolve, reject) => {
+                const realm = new realmConstructor(config);
```

Comment:
Ok maybe this does not work that way. Will need to create an actual local Realm with no schema as the cocoa approach